i18n/create_scratch_msgs.js

```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const assert = require('assert');
+
+// Globals
+const PATH_INPUT = path.resolve(__dirname, '../msg/json/*.json');
+const PATH_OUTPUT = path.resolve(__dirname, '../msg');
+
+let en = fs.readFileSync(path.resolve(__dirname, '../msg/json/en.json'));
+en = JSON.parse(en);
+const enKeys = Object.keys(en).sort().toString();
+
+// Check that translation is valid:
+// entry: array [key, translation]  corresponding to a single string from <locale>.json
+// - messages with placeholders have the same number of placeholders
+// - messages must not have newlines embedded
+const validateEntry = function (entry) {
+    const re = /(%\d)/g;
+    const [key, translation] = entry;
+    const placeholdersCount = en[key].match(re) ? en[key].match(re).length : 0;
+    if (placeholdersCount > 0) {
+      const tMatch = translation.match(re);
+      const enMatch = en[key].match(re);
```

Comment:
`en[key].match(re)` is referenced 3 times here. It can be declared above the `placeholdersCount` declaration.